api/include/opentelemetry/logs/event_id.h

```diff
@@ -0,0 +1,35 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#ifdef ENABLE_LOGS_PREVIEW
+
+#  include "opentelemetry/nostd/unique_ptr.h"
+#  include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace logs
+{
+
+/**
+ * EventId class which acts the Id of the event with an optional name.
+ */
+class EventId
+{
+public:
+  EventId(int64_t id, nostd::string_view name) noexcept
+  {
+    id_   = id;
+    name_ = nostd::unique_ptr<char[]>{new char[name.length() + 1]};
+    std::copy(name.begin(), name.end(), name_.get());
```

Comment:
This goes against the standard practice of not doing any memory copy at the API/SDK surface. The recordable interface ensures that we directly serialize the data at exporter, without creating such intermediate copies. One solution can be to get rid of EventId class, and pass the event-id and event-name directly to Log methods. Also we only use PODs, or `nostd` types as API arguments. Even though we can debate that passing reference/pointer to a user created class mayn't break the ABI, in general good to avoid if we can.

Reply:
> This goes against the standard practice of not doing any memory copy at the API/SDK surface. 

No memory will be allocated in our logging APIs. The `EventId` class is supposed to be created by the user with sort of global lifetime, it so copies the string instead of using any `string_view` on existing `string` because it then requires the user to manage the original string object explicitly. The allocation and free of such object is not in the scope of logging API as it only takes const reference on such object.